### PR TITLE
Enable Cross Compilation of Arm SVE on x86 Cl (examples build and emulation)

### PR DESCRIPTION
**Pull request by rakshithgb-fujitsu**
Based on [#PR2691](https://github.com/oneapi-src/oneDAL/pull/2691) , this update brings in daal and oneapi examples arm build and emulation in a x86 environment.

What's new:

- Both daal and oneapi examples are now available in arm task on the x86 ci. (both modes static and dynamic can now be built using cross-compilation on x86)
- Emulation using qemu for arm is enabled to test the examples.
- `Make file` and `tbb.sh` have both been modified to now handle TBB built from source, avoiding the need to handle the old folder structure.

### CODE REVIEW

**Review by ethanglaser — COMMENTED** [review]
Looks like there are some public CI fails to be addressed

File: .ci/scripts/test.sh [hunk 1]
```diff
@@ -16,45 +16,53 @@
...
+ARCH=${PLATFORM:3:3}
+
 while [[ $# -gt 0 ]]; do
     key="$1"
 
     case $key in
         --test-kind)
         TEST_KIND="$2"
-        ;;
+        shift;;
```
[the remark this answers is not in the document]
Comment (rakshithgb-fujitsu) [reply .ci/scripts/test.sh:30]: It is intended to shift the positional parameters after processing each option-value pair in the loop, also to maintain consistency across all the bash scripts.

**Comment by rakshithgb-fujitsu** [comment]
> Looks like there are some public CI fails to be addressed

I'm not sure what exactly is failing, both the arm and x86 pipelines run both static and dynamic examples successfully and then there is a random bash error that doesn't point to anything specifc. I'm unable to reproduce this in local environment.

@ethanglaser any ideas?

**Comment by keeranroth** [comment]
Hi @rakshithgb-fujitsu I've put up a pull request at https://github.com/oneapi-src/oneDAL/pull/2719 with much of the same changes as this one, but I have some more commits locally based on top of that. It makes the commits for adding LLVM and RISC-V builds easier for me. Would you be able to take a look at that, and if that is OK for you, can we switch to that rather than this one?

**Pull request closed without merging**